src/astx/types/collections.py

```diff
@@ -0,0 +1,92 @@
+"""ASTx collection types module."""
+
+from __future__ import annotations
+
+from typing import List
+
+from public import public
+
+from astx.base import AST, DataType
+from astx.tools.typing import typechecked
+
+
+@public
+@typechecked
+class CollectionType(DataType):
+    """Base class for collection types."""
+
+    pass
+
+
+@public
+@typechecked
+class ListType(CollectionType):
+    """Type representation of a list of elements of a specific type."""
+
+    def __init__(self, element_type: AST) -> None:
```

Comment:
```suggestion
    def __init__(self, element_type: Iterable[ExprType] | ASTNodes[ExprType] ) -> None:
```

it would be necessary to change the line with : `self.elements_type = element_type` to

```python

if isinstance(elements_type, list):
    self.elements_type = ASTNodes[ExprType]()
    ...
```

you can use this as reference: https://github.com/arxlang/astx/blob/main/src/astx/classes.py#L66-L71